### ISSUE TITLE
微调时遇到的问题

### ISSUE THREAD

**Issue by shouji000** [LiAutoAD/LightVLA#19]
大佬您好：
   拜读您的著作“lightvla”后，深感钦佩。便想着深入学习您的工作，在学习过程中，遇到了一些小问题。我在用您提供的指令进行训练时（如下）
torchrun \
  --standalone \
  --nnodes 1 \
  --nproc-per-node 1 \
  vla-scripts/finetune.py \
  --vla_path ./pretrained_checkpoint/openvla-7b \
  --data_root_dir ./libero \
  --dataset_name libero_spatial_no_noops \
  --run_root_dir ./pretrained_checkpoint/CXF-LightVLA-libero-spatial-20251222 \
  --use_l1_regression True \
  --use_diffusion False \
  --use_film False \
  --num_images_in_input 2 \
  --use_proprio True \
  --batch_size 8 \
  --learning_rate 5e-4 \
  --num_steps_before_decay 100000  \
  --max_steps 150005 \
  --save_freq 10000 \
  --save_latest_checkpoint_only False \
  --image_aug True \
  --lora_rank 32
随后进行验证，指令如下
python experiments/robot/libero/run_libero_eval.py --pretrained_checkpoint ./pretrained_checkpoint/CXF-LightVLA-libero-10-20251218/openvla-7b+libero_10_no_noops+b8+lr-0.0005+lora-r32+dropout-0.0--image_aug2025-12-19\ 09\:41\:48.546572--30000_chkpt --task_suite_name libero_10 --save_rollout_video false
出现了下图所示的问题

<img width="985" height="316" alt="Image" src="https://github.com/user-attachments/assets/6399e84b-c4be-44fa-a2a4-0efa4e1e081a" />

模型只留下了极少部分的token，成功率也很低，不知道这是啥问题导致的。

**Comment by Promisery** [comment]
交流群管理起来比较麻烦，有问题的话欢迎在Issue里提，或者给我发邮件也行

**Comment by Promisery** [comment]
> 共做了四次实验，分别是 1.基于openvla-7b，在lightvla框架下，训练了40000步 2.基于openvla-7b，在lightvla框架下，训练了150000步 3.基于openvla-7b-oft-finetuned-libero-spatial（openvla-7b模型利用oft方式在libero-spatial数据上进行微调），在lightvla框架下，训练了150000步 4.基于openvla-7b，在lightvla框架下，训练320000步 其中目前只有第3次实验达到预期效果，如下图
> 
> <img alt="Image" width="1109" height="227" src="https://private-user-images.githubusercontent.com/112758410/531792878-7ab1f9d3-b172-4bb2-9877-d9148d23c098.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Njc3NDY3MzUsIm5iZiI6MTc2Nzc0NjQzNSwicGF0aCI6Ii8xMTI3NTg0MTAvNTMxNzkyODc4LTdhYjFmOWQzLWIxNzItNGJiMi05ODc3LWQ5MTQ4ZDIzYzA5OC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjYwMTA3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI2MDEwN1QwMDQwMzVaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0xOTc3MDVkYTkyNzI5ZGNhNTg3ZWM0ZWI3OTU4ZmYzNDY2YmZkOWZiNmFkNDYzZWY0N2IxYTc0NDRkZjE4Y2I5JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.JXgEkJxH5w8b8POJFyN0PQG80SA798uE99MCNLUluBc">
> 其他三次，测试效果基本上是下图这样的，视觉token会被裁剪的非常多
> 
> <img alt="Image" width="979" height="393" src="https://private-user-images.githubusercontent.com/112758410/531793334-e692d73f-56cb-4c3a-9a15-661b626e9482.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Njc3NDY3MzUsIm5iZiI6MTc2Nzc0NjQzNSwicGF0aCI6Ii8xMTI3NTg0MTAvNTMxNzkzMzM0LWU2OTJkNzNmLTU2Y2ItNGMzYS05YTE1LTY2MWI2MjZlOTQ4Mi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjYwMTA3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI2MDEwN1QwMDQwMzVaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT05NGQ1YzRjMjVhNzljY2ZmM2ZjZmI2NWZiNWQ2MjI5M2E3YWQwZGE1ZTE4NmUxODAxZGZiY2MyMTU5NmExYmQyJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.C8wePtyse8R2XGNatL3QuMuft802u__NljzCVn0VBXs">
> torchrun --standalone --nnodes 1 --nproc-per-node 1 vla-scripts/finetune.py --vla_path ./pretrained_checkpoint/openvla-7b --data_root_dir ./libero --dataset_name libero_spatial_no_noops --run_root_dir ./pretrained_checkpoint/CXF-LightVLA-libero-spatial-20251222 --use_l1_regression True --use_diffusion False --use_film False --num_images_in_input 2 --use_proprio True --batch_size 8 --learning_rate 5e-4 --num_steps_before_decay 30000 --max_steps 40005 --save_freq 10000 --save_latest_checkpoint_only False --image_aug True --lora_rank 32
> 
> torchrun --standalone --nnodes 1 --nproc-per-node 1 vla-scripts/finetune.py --vla_path ./pretrained_checkpoint/openvla-7b --data_root_dir ./libero --dataset_name libero_spatial_no_noops --run_root_dir ./pretrained_checkpoint/CXF-LightVLA-libero-spatial-20251222 --use_l1_regression True --use_diffusion False --use_film False --num_images_in_input 2 --use_proprio True --batch_size 8 --learning_rate 5e-4 --num_steps_before_decay 100000 --max_steps 150005 --save_freq 10000 --save_latest_checkpoint_only False --image_aug True --lora_rank 32
> 
> torchrun --standalone --nnodes 1 --nproc-per-node 1 vla-scripts/finetune.py --vla_path ./pretrained_checkpoint/openvla-7b-oft-finetuned-libero-spatial --data_root_dir ./libero --dataset_name libero_spatial_no_noops --run_root_dir ./pretrained_checkpoint/CXF-LightVLA-libero-spatial-20251225 --use_l1_regression True --use_diffusion False --use_film False --num_images_in_input 2 --use_proprio True --batch_size 8 --learning_rate 5e-4 --num_steps_before_decay 100000 --max_steps 150005 --save_freq 10000 --save_latest_checkpoint_only False --image_aug True --lora_rank 32
> 
> torchrun --standalone --nnodes 1 --nproc-per-node 1 vla-scripts/finetune.py --vla_path ./pretrained_checkpoint/openvla-7b --data_root_dir ./libero --dataset_name libero_spatial_no_noops --run_root_dir ./pretrained_checkpoint/CXF-LightVLA-libero-spatial-20251229 --use_l1_regression True --use_diffusion False --use_film False --num_images_in_input 2 --use_proprio True --batch_size 8 --learning_rate 5e-4 --num_steps_before_decay 200000 --max_steps 320005 --save_freq 10000 --save_latest_checkpoint_only False --image_aug True --lora_rank 32
> 
> 大佬可以帮我解答一下这个疑问嘛

在训练LightVLA的时候，请使用已经微调过的OFT checkpoint，可以从[这里](https://huggingface.co/moojink/models?search=oft)找到，不要用OpenVLA checkpoint。

LightVLA本身是比较容易出现模式坍塌的，也就是你看到的每次只选出来几个token，模型需要有一个比较好的基础能力才能保证LightVLA不坍塌，所以需要用已经微调过的OFT checkpoint，用OpenVLA checkpoint就很容易爆。

另外，就像我之前说的，请加入`--grad_accumulation_steps 8`参数来让batch size和我们的设置保持一致。

**Comment by may210297** [comment]
#23 可以参考这个pr，谢谢关注！
